File: java/src/test/java/mx/iteso/CommenterTest.java
package mx.iteso;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;
import static org.junit.Assert.assertFalse;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

import org.junit.Before;
import org.junit.Test;

import mx.iteso.Commenter;

public class CommenterTest {
    private User user;

    @Before
    public void setUp() {
        user = mock((User.class));
    }

    /** Test de permiso de aprobación. */
    @Test
    public void testApprovePermission() {
        User commenter = new Commenter(this.user);
        when(this.user.approve()).thenReturn(false);
        assertFalse(commenter.approve());
    }

    /** Test de permiso para realizar comentarios. */
    @Test
    public void testCommenterPermission() {
        User commenter = new Commenter(this.user);
        when(this.user.comment()).thenReturn(false);
        assertTrue(commenter.comment());
    }
    /** Test de permiso para eliminar. */
    @Test
    public void testDeleterPermission() {
        User commenter = new Commenter(this.user);
        when(this.user.delete()).thenReturn(false);
        assertFalse(commenter.delete());
    }
    /** Test de permiso para publicar. */
    @Test
    public void testPublisherPermission() {
        User commenter = new Commenter(this.user);
        when(this.user.publish()).thenReturn(false);
        assertFalse(commenter.publish());
    }
    /** Test de la descripción del usuario. */
    @Test
    public void testDescription() {
        User commenter = new Commenter(user);
        when(user.getDescription()).thenReturn("User >");
        assertEquals("User > Can comment", commenter.getDescription());
    }
}

File: python/src/unittest/python/CommenterTest.py
from mockito import mock, verify, when
import unittest
from User import *
from Commenter import *


class CommenterTest(unittest.TestCase):
    def setUp(self):
        self.user = mock(User, strict=False)

    def approveTest(self):
        commenter = Commenter(self.user)
        when(self.user).approve().thenReturn(False)
        self.assertFalse(commenter.approve())

    def commenterTest(self):
        commenter = Commenter(self.user)
        when(self.user).comment().thenReturn(False)
        self.assertTrue(commenter.comment())

    def deleterTest(self):
        commenter = Commenter(self.user)
        when(self.user).delete().thenReturn(False)
        self.assertFalse(commenter.delete())

    def publisherTest(self):
        commenter = Commenter(self.user)
        when(self.user).publish().thenReturn(False)
        self.assertFalse(commenter.publish())

    def getDescriptionTest(self):
        commenter = Approval(self.user)
        when(self.user).getDescription().thenReturn("User >")
        self.assertEquals("User > Can comment", commenter.getDescription())
